c++/src/kj/refcount.h

```diff
@@ -115,6 +142,102 @@ Own<T> Refcounted::addRefInternal(T* object) {
   return Own<T>(object, *refcounted);
 }
 
+template <typename T>
+Rc<T> Refcounted::addRcRefInternal(T* object) {
+  Refcounted* refcounted = object;
+  ++refcounted->refcount;
+  return Rc<T>(object);
+}
+
+template<typename T>
+class Rc: public Own<T> {
```

Comment:
I'm generally not a big fan of `kj::Rc<T>` extending from `kj::Own<T>`. Yes it makes migration easier but also makes it easy to miss and get things wrong.